### PR TITLE
Add init container to GFD for handling imex nodes config mount

### PR DESCRIPTION
**Pull request by cdesiniotis**

### CODE REVIEW

File: assets/gpu-feature-discovery/0500_daemonset.yaml [hunk 1]
```diff
@@ -36,6 +36,47 @@ spec:
...
+              echo "Copying IMEX nodes config"
+              mkdir -p $(dirname /config/${IMEX_NODES_CONFIG_FILE})
+              cp ${DRIVER_ROOT_CTR_PATH}/${IMEX_NODES_CONFIG_FILE} /config/${IMEX_NODES_CONFIG_FILE}
+          volumeMounts:
+            - name: config
+              mountPath: /config
+            - name: run-nvidia
+              mountPath: /run/nvidia
+              readOnly: true
+              mountPropagation: Bidirectional
```
Comment (klueska) [inline assets/gpu-feature-discovery/0500_daemonset.yaml]: Why does this one need the ability to do bi-directional mounts?

File: assets/gpu-feature-discovery/0500_daemonset.yaml [hunk 2]
```diff
@@ -36,6 +36,47 @@ spec:
...
+              set -o allexport
+              cat /run/nvidia/validations/driver-ready
+              . /run/nvidia/validations/driver-ready
+              
+              IMEX_NODES_CONFIG_FILE=/etc/nvidia-imex/nodes_config.cfg
+              if [[ -f /config/${IMEX_NODES_CONFIG_FILE} ]]; then
+                echo "Removing cached IMEX nodes config"
+                rm -f /config/${IMEX_NODES_CONFIG_FILE}
+              fi
+              if [[ ! -f ${DRIVER_ROOT_CTR_PATH}/${IMEX_NODES_CONFIG_FILE} ]]; then
```
Comment (klueska) [inline assets/gpu-feature-discovery/0500_daemonset.yaml:58]: Where is DRIVER_ROOT_CTR_PATH being set?

Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml:58]: A few lines above when we source the `driver-ready` file.

File: assets/gpu-feature-discovery/0500_daemonset.yaml (hunk 1 above)
Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml]: I don't believe this is strictly required. I set this to `Bidirectional` to remain consistent with how we mount this volume in the `toolkit-validation` init container on line 38.

Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml]: @tariq1890 recently removed the need to run our `toolkit-validation` init containers as `privileged` (in addition to changing the mount propagation of the `/run/nvidia` mount from `Bidirectional` to `HostToContainer`) https://github.com/NVIDIA/gpu-operator/commit/dc1ea09ac9444455fc4327897dc09f950198ae66. But we rolled back that change after discovering a regression on OpenShift / RHEL based clusters where SELinux is enabled and enforcing: https://github.com/NVIDIA/gpu-operator/pull/1034

**Comment by cdesiniotis** [comment]
@ArangoGutierrez @tariq1890 do you know why the GitHub actions are not being triggered here?

File: assets/gpu-feature-discovery/0500_daemonset.yaml (hunk 1 above)
Comment (tariq1890) [reply assets/gpu-feature-discovery/0500_daemonset.yaml]: @cdesiniotis Just to clarify, RHEL with SELinux required privileged mode even for `HostToContainer`, whereas privileged containers were only mandated for `Bidirectional` in conventional (non-SELinux) environments.

We don't need `Bidirectional` mounts to satisfy the SELinux constraint, so I think we should just stick to the `HostToContainer` mountPropagation

Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml]: Ack. I'll update the mount to `HostToContainer` in this PR.

Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml:71]: Ack. I'll update the mount to `HostToContainer` in this PR.

**Comment by tariq1890** [comment]
> @ArangoGutierrez @tariq1890 do you know why the GitHub actions are not being triggered here?

Opened PR https://github.com/NVIDIA/gpu-operator/pull/1071 to fix this

File: assets/gpu-feature-discovery/0500_daemonset.yaml (hunk 1 above)
Comment (tariq1890) [reply assets/gpu-feature-discovery/0500_daemonset.yaml]: Thanks!

File: assets/gpu-feature-discovery/0500_daemonset.yaml [hunk 3]
```diff
@@ -156,3 +183,5 @@ spec:
           hostPath:
             path: /run/nvidia/driver
             type: DirectoryOrCreate
+        - name: config
+          emptyDir: {}
```
Comment (tariq1890) [inline assets/gpu-feature-discovery/0500_daemonset.yaml]: EOL here please

Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml]: updated.

File: assets/gpu-feature-discovery/0500_daemonset.yaml (hunk 2 above)
Comment (cdesiniotis) [reply assets/gpu-feature-discovery/0500_daemonset.yaml:58]: Here are the logs from the init container now:
```
cdesiniotis@cnt-imex-devel:~$ kubectl logs -n gpu-operator gpu-feature-discovery-5nd22 -c gpu-feature-discovery-imex-init
IS_HOST_DRIVER=true
NVIDIA_DRIVER_ROOT=/
DRIVER_ROOT_CTR_PATH=/host
NVIDIA_DEV_ROOT=/
DEV_ROOT_CTR_PATH=/host
Copying IMEX nodes config
```

**Review by tariq1890 — APPROVED** [review]

**Pull request merged**